src/java/org/apache/cassandra/service/accord/txn/AbstractSerialized.java

```diff
@@ -84,8 +109,42 @@ public void unmemoize()
         memoized = null;
     }
 
-    protected ByteBuffer bytes()
+    @Nullable
+    protected ByteBuffer unsafeBytes()
     {
         return bytes;
     }
+
+    @Nonnull
+    protected ByteBuffer bytes(Version target)
+    {
+        Invariants.nonNull(bytes);
```

Comment:
its nullable in a few contexts, but not in any context that *needs* the bytes, so this is here to enforce that behavior.